src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+        linear (P1).
     """
 
-    def __init__(self, nt, solver_options='operator'):
+    IteratorType = ImplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, solver_options='operator', interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        return implicit_euler(operator, rhs, mass, initial_data, initial_time, end_time, self.nt, mu, num_values,
-                              solver_options=self.solver_options)
+
+class ExplicitEulerIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times)
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the step function U_np1 = M^{-1}(M U_n + dt F - dt A U_n)
+        if not 't' in M.parameters:
+            M = M.assemble(mu)
+        if not 't' in A.parameters:
+            A = A.assemble(mu)
+
+        if isinstance(F, ZeroOperator):
+            if isinstance(M, IdentityOperator):
+                def step_function(U_n, t_n):
+                    t_np1 = t_n + dt
+                    mu_t = mu.with_(t=t_np1)
+                    return U_n - dt*A.apply(U_n, mu=mu_t), t_np1
+            else:
+                def step_function(U_n, t_n):
+                    t_np1 = t_n + dt
+                    mu_t = mu.with_(t=t_np1)
+                    return (
+                        M.apply_inverse(M.apply(U_n, mu=mu_t) - dt*A.apply(U_n, mu=mu_t), mu=mu_t, initial_guess=U_n),
+                        t_np1)
+        else:
+            if not F.parametric or 't' not in F.parameters:
+                F = F.assemble(mu)
+            if isinstance(M, IdentityOperator):
+                def step_function(U_n, t_n):
+                    t_np1 = t_n + dt
+                    mu_t = mu.with_(t=t_np1)
+                    return U_n + dt*(F.as_vector(mu_t) - A.apply(U_n, mu=mu_t)), t_np1
+            else:
+                def step_function(U_n, t_n):
+                    t_np1 = t_n + dt
+                    mu_t = mu.with_(t=t_np1)
+                    return (
+                        M.apply_inverse(M.apply(U_n, mu=mu_t) + dt(F.as_vector(mu_t) - A.apply(U_n, mu=mu_t)),
+                                        mu=mu_t, initial_guess=U_n),
+                        t_np1)
+
+        self.step_function = step_function
+
+    def _step_function(self, U_n, t_n):
+        return self.step_function(U_n, t_n)
 
 
 class ExplicitEulerTimeStepper(TimeStepper):
```

Comment:
Do we really still need this one, given that we have a general explicit RK stepper?

Reply:
Not really, I guess. Could be that its a bit faster than the general RK one, though. Want to drop it?

Reply:
Would probably good to first check, how performance is actually affected. Maybe an argument for keeping it would be that if you somehow want to tweak timestepping, this code might be easier to work with.

Reply:
I don't have a very strong opinion on this. You decide ..